### PR TITLE
refactor: use inject api instead of constructor for DI

### PR DESCRIPTION
**Pull request by chintankavathia**
**What kind of change does this PR introduce?** (check one with "x")
Switch to `inject` syntax instead of using constructor style DI 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/ngx-datatable/src/lib/directives/orderable.directive.spec.ts
Comment (timowolf) [inline projects/ngx-datatable/src/lib/directives/orderable.directive.spec.ts:1]: This change of the test file seems not related. Is it intended?

**Review by timowolf — CHANGES_REQUESTED** [review]
Looks good. Many thanks, just one question?

File: projects/ngx-datatable/src/lib/directives/orderable.directive.spec.ts
Comment (chintankavathia) [reply projects/ngx-datatable/src/lib/directives/orderable.directive.spec.ts:1]: It is required as it was wrongly creating directive instances using constructor calls before. now with this migration angular will throw error when inject gets called out of injector context.

File: projects/ngx-datatable/src/lib/components/body/body-row.component.ts [hunk 1]
```diff
@@ -61,6 +61,9 @@ import { ColumnGroupWidth, PinnedColumns } from '../../types/internal.types';
   `
 })
 export class DataTableBodyRowComponent<TRow = any> implements DoCheck, OnChanges {
+  private scrollbarHelper = inject(ScrollbarHelper, { skipSelf: true });
```
Comment (spike-rabbit) [inline projects/ngx-datatable/src/lib/components/body/body-row.component.ts:64]:
```suggestion
  private scrollbarHelper = inject(ScrollbarHelper);
```

`ScrollbarHelper` is provided in the module. So for me it looks like skipSelf is pointless and we can just remove it

File: projects/ngx-datatable/src/lib/components/columns/column-cell.directive.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import { Directive, TemplateRef } from '@angular/core';
+import { Directive, inject, TemplateRef } from '@angular/core';
 import { CellContext } from '../../types/public.types';
 
 @Directive({ selector: '[ngx-datatable-cell-template]' })
 export class DataTableColumnCellDirective {
-  constructor(public template: TemplateRef<any>) {}
+  template = inject<TemplateRef<any>>(TemplateRef);
```
Comment (spike-rabbit) [inline projects/ngx-datatable/src/lib/components/columns/column-cell.directive.ts:6]:
```suggestion
  template = inject<TemplateRef<CellContext>>(TemplateRef);
```

File: projects/ngx-datatable/src/lib/components/datatable.component.ts [hunk 3]
```diff
@@ -85,6 +83,12 @@ import {
 export class DatatableComponent<TRow = any>
   implements OnInit, DoCheck, AfterViewInit, AfterContentInit, OnDestroy
 {
+  private scrollbarHelper = inject(ScrollbarHelper, { skipSelf: true });
+  private dimensionsHelper = inject(DimensionsHelper, { skipSelf: true });
```
Comment (spike-rabbit) [inline projects/ngx-datatable/src/lib/components/datatable.component.ts:87]:
```suggestion
  private scrollbarHelper = inject(ScrollbarHelper);
  private dimensionsHelper = inject(DimensionsHelper);
```

Both are provided in the module. So for me it looks like skipSelf is pointless and we can just remove it

File: projects/ngx-datatable/src/lib/components/header/header.component.ts [hunk 4]
```diff
@@ -92,6 +92,9 @@ import {
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class DataTableHeaderComponent implements OnDestroy, OnChanges {
+  private cd = inject(ChangeDetectorRef);
+  private scrollbarHelper = inject(ScrollbarHelper, { skipSelf: true });
```
Comment (spike-rabbit) [inline projects/ngx-datatable/src/lib/components/header/header.component.ts:96]:
```suggestion
  private scrollbarHelper = inject(ScrollbarHelper);
```

**Review by spike-rabbit — CHANGES_REQUESTED** [review]
Looks good, just a few things I noticed that were also wrong before but related

**Review by timowolf — APPROVED** [review]
LGTM

**Pull request merged**